algorithms/src/snark/marlin/data_structures/proof.rs

```diff
@@ -318,3 +369,152 @@ impl<E: PairingEngine> FromBytes for Proof<E> {
         Self::deserialize_compressed(&mut r).map_err(|_| error("could not deserialize Proof"))
     }
 }
+
+#[cfg(test)]
+mod test {
+    #![allow(non_camel_case_types)]
+
+    use super::*;
+
+    use crate::{
+        polycommit::{
+            kzg10::{KZGCommitment, KZGProof},
+            sonic_pc::BatchProof,
+        },
+        snark::marlin::prover::MatrixSums,
+    };
+    use snarkvm_curves::{
+        bls12_377::{Bls12_377, Fr, G1Affine},
+        AffineCurve,
+    };
+    use snarkvm_utilities::{TestRng, Uniform};
+
+    fn modes() -> [(Compress, Validate); 4] {
+        [
+            (Compress::No, Validate::No),
+            (Compress::Yes, Validate::No),
+            (Compress::No, Validate::Yes),
+            (Compress::Yes, Validate::Yes),
+        ]
+    }
+
+    fn sample_commit() -> KZGCommitment<Bls12_377> {
+        let buf = G1Affine::prime_subgroup_generator().to_bytes_le().unwrap();
+        FromBytes::read_le(buf.as_slice()).unwrap()
+    }
+
+    fn rand_commitments(i: usize, j: usize, test_with_none: bool) -> Commitments<Bls12_377> {
+        assert!(i > 0);
+        assert!(j > 0);
+        let mask_poly = if test_with_none { None } else { Some(sample_commit()) };
```

Comment:
don't know how "heavy" ``sample_commit` is, but perhaps it would be a good idea to create it just once and then clone it around?

Reply:
A single G1Affine, I think we'll survive: cef8f291b